extensions/shared/components/audio-player/index.jsx

```diff
@@ -0,0 +1,160 @@
+/* global _wpmejsSettings, MediaElementPlayer */
+
+/**
+ * External dependencies
+ */
+import { throttle } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { STATE_PLAYING, STATE_PAUSED, STATE_ERROR } from './constants';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const meJsSettings = typeof _wpmejsSettings !== 'undefined' ? _wpmejsSettings : {};
+
+function createJumpButton( containerClass, label, clickHandler ) {
+	const buttonContainer = document.createElement( 'div' );
+	buttonContainer.className = containerClass;
+
+	const button = document.createElement( 'button' );
+	button.innerText = label;
+	button.addEventListener( 'click', clickHandler );
+	buttonContainer.appendChild( button );
+	return buttonContainer;
+}
+
+function AudioPlayer( {
+	trackSource,
+	onPlay,
+	onPause,
+	onError,
+	onTimeChange,
+	onSkipForward,
+	onJumpBack,
+	currentTime,
+	playStatus = STATE_PAUSED,
+} ) {
+	const audioRef = useRef();
+
+	/**
+	 * Play current audio.
+	 *
+	 * @public
+	 */
+	const play = () => {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		audioRef.current.play().catch( () => {} );
```

Comment:
could be possible that `current` is `undefined`? If so, maybe we should check before to use its properties

```suggestion
		audioRef.current?.play().catch( () => {} );
```

Reply:
It shouldn't be possible, as these `play` and `pause` functions are only ever called inside of this `useEffect` hook:

https://github.com/Automattic/jetpack/blob/update/audio-player/extensions/shared/components/audio-player/index.jsx#L112-L122

The audio player status can only change if it exists. If the player status is out of sync with the prop, that will cause one of these to functions to be called. You could do something odd and pass in a `playing` status at the start, which would mean we might end up calling this on a null value, but should we guard against that?